loadtest/clan.go

```diff
@@ -0,0 +1,80 @@
+package loadtest
+
+import (
+	"math/rand"
+
+	"github.com/topfreegames/khan/lib"
+)
+
+func (app *App) setClanConfigurationDefaults() {
+}
+
+func (app *App) configureClanOperations() {
+	app.appendOperation(app.getUpdateSharedClanScoreOperation())
+}
+
+func (app *App) getUpdateSharedClanScoreOperation() operation {
+	operationKey := "updateSharedClanScore"
+	return operation{
+		probability: app.getOperationProbabilityConfig(operationKey),
+		canExecute: func() (bool, error) {
+			count, err := app.cache.getSharedClansCount()
+			if err != nil {
+				return false, err
+			}
+			return count > 0, nil
+		},
+		execute: func() error {
+			err := app.loadSharedClansMembers() // cached
+			if err != nil {
+				return err
+			}
+
+			clanPublicID, playerPublicID, err := app.cache.chooseRandomSharedClanAndPlayer()
+			if err != nil {
+				return err
+			}
+
+			// updatePlayer
+			playerMetadata := make(map[string]interface{})
+			playerMetadata["score"] = int(rand.Float64() * 1000)
+			_, err = app.client.UpdatePlayer(nil, playerPublicID, "PlayerName", playerMetadata)
+			if err != nil {
+				return err
+			}
+
+			// getClan
+			clan, err := app.client.RetrieveClan(nil, clanPublicID)
+			if err != nil {
+				return err
+			}
+
+			// updateClan
+			clanScore := getPlayerScoreFromMetadata(clan.Owner.Metadata)
+			for _, member := range clan.Roster {
+				clanScore += getPlayerScoreFromMetadata(member.Player.Metadata)
+			}
+			clanMetadata := make(map[string]interface{})
+			clanMetadata["score"] = clanScore
```

Comment:
you're overwritting the clan score on purpose?

Reply:
Yes, just like Castle Crush.

Reply:
okay, we don't actually overwrite it with the player's score but rather a new clan score, but for testing purposes this is good enough

Reply:
I've actually implemented almost the same formula for clan score used in Castle Crush, except that I'm not using weights for the five sorted chunks of 10 players inside the clan.